libcudacxx/include/cuda/__floating_point/operators.h

```diff
@@ -0,0 +1,1250 @@
+
+      const auto __l = static_cast<_CommonFp>(__lhs);
+      const auto __r = static_cast<_CommonFp>(__rhs);
+
+      if (!_CUDA_VSTD::__cccl_default_is_constant_evaluated())
+      {
+        NV_IF_TARGET(NV_IS_DEVICE, (return __gt_impl_device(__l, __r);))
+      }
+
+      return __gt_impl_constexpr(__l, __r);
+    }
+  }
+
+  /********************************************************************************************************************/
+  // Greater than or equal
+  /********************************************************************************************************************/
+  template <class _Tp>
+  _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI constexpr static bool
+  __ge_impl_constexpr(const _Tp& __lhs, const _Tp& __rhs) noexcept
+  {
+    if (__lhs.__is_nan() || __rhs.__is_nan())
+    {
+      return false;
+    }
+
+    return (__lhs.__storage_ & __lhs.__mask()) >= (__rhs.__storage_ & __rhs.__mask());
+  }
+
+  template <class _Tp>
+  _CCCL_NODISCARD _CCCL_HIDE_FROM_ABI _CCCL_DEVICE static bool
+  __ge_impl_device(const _Tp& __lhs, const _Tp& __rhs) noexcept
+  {
+#  if __cccl_ptx_isa >= 650
+    if constexpr (_CUDA_VSTD::_CCCL_TRAIT(is_same, _Tp, fp16))
+    {
+      NV_IF_TARGET(NV_PROVIDES_SM_53,
+                   (_CUDA_VSTD::uint16_t __ret;
+                    asm("set.ge.u16.f16 %0, %1, %2;" : "=h"(__ret) : "h"(__lhs.__storage_), "h"(__rhs.__storage_));
+                    return static_cast<bool>(__ret);))
+    }
+#  endif // __cccl_ptx_isa >= 650
+#  if __cccl_ptx_isa >= 780
+    if constexpr (_CUDA_VSTD::_CCCL_TRAIT(is_same, _Tp, bf16))
+    {
+      NV_IF_TARGET(NV_PROVIDES_SM_90,
+                   (_CUDA_VSTD::uint16_t __ret;
+                    asm("set.ge.u16.bf16 %0, %1, %2;" : "=h"(__ret) : "h"(__lhs.__storage_), "h"(__rhs.__storage_));
+                    return static_cast<bool>(__ret);))
+    }
+#  endif // __cccl_ptx_isa >= 780
+
+    return __ge_impl_constexpr(__lhs, __rhs);
+  }
+
+  template <class _Lhs, class _Rhs>
+  _CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI static constexpr auto __ge(const _Lhs& __lhs, const _Rhs& __rhs) noexcept
+  {
+    if constexpr (_Lhs::__has_native_type() && _Rhs::__has_native_type())
+    {
+      NV_IF_ELSE_TARGET(NV_IS_HOST,
+                        (return __lhs.__host_native() >= __rhs.__host_native();),
+                        (return __lhs.__device_native() >= __rhs.__device_native();))
+    }
+    else
+    {
+      using _CommonFp = __fp_common_type_t<_Lhs, _Rhs>;
+
+      const auto __l = static_cast<_CommonFp>(__lhs);
+      const auto __r = static_cast<_CommonFp>(__rhs);
+
+      if (!_CUDA_VSTD::__cccl_default_is_constant_evaluated())
+      {
+        NV_IF_TARGET(NV_IS_DEVICE, (return __ge_impl_device(__l, __r);))
+      }
+
+      return __ge_impl_constexpr(__l, __r);
+    }
+  }
+};
+
+/**********************************************************************************************************************/
+// Unary operators
```

Comment:
All of these operators should be hidden friends of the `__fp` template. This is creating a whole lot of operator overloads, for some very common operators, and we should do our best to hide them unless they are in any way relevant.

Reply:
I agree with Michal on this one.  Making them hidden friends is important.